src/awkward/_connect/pyarrow.py

```diff
@@ -944,6 +951,17 @@ def handle_arrow(obj, generate_bitmasks=False, pass_empty_field=False):
 
         if len(layouts) == 1:
             return layouts[0]
+        elif any(is_revertable(arr) for arr in layouts):
+            assert all(is_revertable(arr) for arr in layouts)
```

Comment:
All of the arrays in an Arrow ChunkedArray will have the same data type; that's either guaranteed at the level of the Table schema or (if a ChunkedArray can live on its own, outside of a Table...?) in some ChunkedArray type attribute.

I just verified it: a ChunkedArray has a `type` attribute, and it's pretty strict:

```python
>>> import pyarrow as pa

>>> pa.chunked_array([pa.array([[1, 2, 3], [], [4, 5]]), pa.array([[1]])]).type
ListType(list<item: int64>)

>>> pa.chunked_array([pa.array([[1, 2, 3], [], [4, 5]]), pa.array([[1.1]])]).type
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/table.pxi", line 1327, in pyarrow.lib.chunked_array
  File "pyarrow/error.pxi", line 144, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 123, in pyarrow.lib.check_status
pyarrow.lib.ArrowTypeError: Array chunks must all be same type
```

So this assertion will always pass, but it's good to keep it in, since it explains the code on the next few lines.